zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Follower.java

```diff
@@ -248,14 +249,9 @@ protected void processPacket(QuorumPacket qp) throws Exception {
      * @return zxid
      */
     public long getZxid() {
-        try {
-            synchronized (fzk) {
-                return fzk.getZxid();
-            }
-        } catch (NullPointerException e) {
```

Comment:
I wish we could find out why we try to catch NPE here--has it been thrown before? and it just prints a warning msg and doesn't call shutdown or anything.

Reply:
I'm not sure on the history there... maybe someone at some time was passing a null here.  Like I said, that is no longer the case and I am enforcing it in the constructor.

Reply:
It's being there since the very beginning of ZK project, might be a quick hack done at that time.

It's safe to remove it given we'll always pass in the FollowerZooKeeperServer when initializing Follower.